### PR TITLE
API 600,800,1000 and 1200 support for Server hardware type and API 800 to 1200 Support for Enclosure Group

### PR DESCRIPTION
**Pull request by AnushaY1916**
### Description
API 600,800,1000 and 1200 support for Server hardware type and API 800 to 1200 Support for Enclosure Group

### Issues Resolved
NA

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [ ] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: libraries/resource_providers/api600.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# (c) Copyright 2018 Hewlett Packard Enterprise Development LP
+# (c) Copyright 2020 Hewlett Packard Enterprise Development LP
```
Comment (soodpr) [inline libraries/resource_providers/api600.rb]: this is not required as we are not changing any code here

File: libraries/resource_providers/api600/c7000.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# (c) Copyright 2018 Hewlett Packard Enterprise Development LP
+# (c) Copyright 2020 Hewlett Packard Enterprise Development LP
```
Comment (soodpr) [inline libraries/resource_providers/api600/c7000.rb]: this is not required as we are not changing any code here

File: libraries/resource_providers/api600/synergy.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# (c) Copyright 2018 Hewlett Packard Enterprise Development LP
+# (c) Copyright 2020 Hewlett Packard Enterprise Development LP
```
Comment (soodpr) [inline libraries/resource_providers/api600/synergy.rb]: same as above

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,10 +1,13 @@
-## 3.1.1(unreleased)
-Extending support to API 600 with the following HPE OneView resource:
-- oneview_scope
+## 4.0.0(unreleased)
```
Comment (soodpr) [inline CHANGELOG.md]: i think it should be 3.2.0

Comment (AnushaY1916) [reply CHANGELOG.md]: Updated

File: libraries/resource_providers/api600.rb (hunk 1 above)
Comment (AnushaY1916) [reply libraries/resource_providers/api600.rb]: Updated

File: libraries/resource_providers/api600/c7000.rb (hunk 2 above)
Comment (AnushaY1916) [reply libraries/resource_providers/api600/c7000.rb]: Updated

File: libraries/resource_providers/api600/synergy.rb (hunk 3 above)
Comment (AnushaY1916) [reply libraries/resource_providers/api600/synergy.rb]: Updated

**Review by soodpr — APPROVED** [review]
LGTM 👍

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,10 +1,14 @@
-## 3.1.1(unreleased)
-Extending support to API 600 with the following HPE OneView resource:
-- oneview_scope
+## 3.2.0(unreleased)
+### Notes
```
Comment (jyothisgm) [inline CHANGELOG.md]: Do not remove the changelog of the previous unreleased version. Please append it to the current one.

File: libraries/resource_providers/api1000.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# (c) Copyright 20202020202020202020 Enterprise Development LP
```
Comment (jyothisgm) [inline libraries/resource_providers/api1000.rb]: Please correct this line.

File: libraries/resource_providers/api800/synergy/server_hardware_type_provider.rb [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
```
Comment (jyothisgm) [inline libraries/resource_providers/api800/synergy/server_hardware_type_provider.rb]: Correct the year

File: libraries/resource_providers/api800/synergy.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# (c) Copyright 2018 Hewlett Packard Enterprise Development LP
```
Comment (jyothisgm) [inline libraries/resource_providers/api800/synergy.rb]: Copyright year.

File: libraries/resource_providers/api800/c7000/server_hardware_type_provider.rb [hunk 9]
```diff
@@ -0,0 +1,20 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
```
Comment (jyothisgm) [inline libraries/resource_providers/api800/c7000/server_hardware_type_provider.rb]: Correct copyright year

**Review by jyothisgm — CHANGES_REQUESTED** [review]

File: libraries/resource_providers/api1000.rb (hunk 6 above)
Comment (AnushaY1916) [reply libraries/resource_providers/api1000.rb]: Updated

File: CHANGELOG.md (hunk 5 above)
Comment (AnushaY1916) [reply CHANGELOG.md]: Done

File: libraries/resource_providers/api800/c7000/server_hardware_type_provider.rb (hunk 9 above)
Comment (AnushaY1916) [reply libraries/resource_providers/api800/c7000/server_hardware_type_provider.rb]: Corrected

File: libraries/resource_providers/api800/synergy.rb (hunk 8 above)
Comment (AnushaY1916) [reply libraries/resource_providers/api800/synergy.rb]: Corrected

File: libraries/resource_providers/api800/synergy/server_hardware_type_provider.rb (hunk 7 above)
Comment (AnushaY1916) [reply libraries/resource_providers/api800/synergy/server_hardware_type_provider.rb]: Corrected

File: CHANGELOG.md [hunk 10]
```diff
@@ -84,7 +86,7 @@ Adds API 500 support to the following HPE OneView resources:
   - oneview_server_hardware_type
   - oneview_server_profile
   - oneview_server_profile_template
-  - oneview_storage_pool
+  - oneview_storage_p
```
Comment (jyothisgm) [inline CHANGELOG.md]: THis got cut. Please fix.

Comment (AnushaY1916) [reply CHANGELOG.md]: Fixed it

**Review by jyothisgm — APPROVED** [review]
LGTM

**Pull request merged**